src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +240,206 @@ def truncate_table(self, table):
 
+    def is_native_autodetect_schema_available(self, file: File) -> bool:
+        """
+        Check if native auto-detection of schema is available.
+
+        :param file: File used to check the file type of to decide whether there is a native auto-detection available.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if supported_config and file.type.name in supported_config["filetype"]:  # type: ignore
+            return True
+        return False
+
+    def create_table_using_native_schema_autodetection(
+        self,
+        table: Table,
+        file: File,
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file via native database support.
+
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if not supported_config:
+            raise ValueError("Unsupported file location.")
+        return supported_config["method"](table=table, file=file)  # type: ignore
+
+    def is_native_path_available(self, source_dataset: File | Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_dataset: File from which we need to transfer data
+        """
+        if isinstance(source_dataset, Table):
+            return False
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_dataset.type.name)
+        location_type = self.NATIVE_PATHS.get(source_dataset.location.location_type)
+        return bool(location_type and file_type)
+
+    def load_file_to_table_natively(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        """
+        method_name = self.NATIVE_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+        else:
+            raise DatabaseCustomError(
+                f"No transfer performed since there is no optimised path "
+                f"for {source_file.location.location_type} to bigquery."
+            )
```

Comment:
```suggestion
        if not method_name:
            raise DatabaseCustomError(
                f"No transfer performed since there is no optimised path "
                f"for {source_file.location.location_type} to bigquery."
            )

        transfer_method = self.__getattribute__(method_name)
        transfer_method(
            source_file=source_file,
            target_table=target_table,
            if_exists=if_exists,
            **kwargs,
        )

```